### PR TITLE
Pyrodigal migrate to nf-test

### PR DESCRIPTION
**Pull request by Darcy220606**
This PR updates the Pyrodigal module to nf-test. Thanks to @louperelo 🥇 

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by louperelo — APPROVED** [review]
looks good to me and all tests passed.

File: modules/nf-core/pyrodigal/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+}
```
Comment (mashehu) [inline modules/nf-core/pyrodigal/tests/main.nf.test:30]: Please also add a stub test, which we will probably require soon for nf-core modules.
You can see an example here: https://github.com/nf-core/modules/pull/4572/files

**Review by mashehu — APPROVED** [review]

File: modules/nf-core/pyrodigal/tests/main.nf.test (hunk 1 above)
Comment (Darcy220606) [reply modules/nf-core/pyrodigal/tests/main.nf.test:30]: @mashehu But i thought a stub test is required only if a module relies on large databases for the test data like mentioned in the [DSL2 documentation](https://nf-co.re/docs/contributing/modules#migrating-from-pytest-to-nf-test). Pyrodigal doesnt need such large test data.

Comment (mashehu) [reply modules/nf-core/pyrodigal/tests/main.nf.test:30]: One use case for stubs is as you describe, if we can't run the normal tests. Another one is to quickly go through a whole pipline as outlined here: https://github.com/nf-core/modules/issues/4570
It's fine as it is, but might change soon™️

Comment (Darcy220606) [reply modules/nf-core/pyrodigal/tests/main.nf.test:56]: Hey @mashehu . Thanks for the explanation. 
In that case i just added the stub section to pyrodigal. Would you please have a quick look again? Much appreciated.

**Review by mashehu — APPROVED** [review]

**Pull request merged**